### PR TITLE
feat(component/home+search): Adding home and search component with ro…

### PR DESCRIPTION
**Pull request by jpukg**
…uting enabled

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/web-slate/simple-google-search/blob/main/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What type of change does this PR introduce?

- [ ] API
- [x] Frontend

## PR Sub Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: #24 

## Does this PR introduce a breaking change?

- [x] Yes
- [ ] No

## Other information

### CODE REVIEW

**Comment by jpukg** [comment]
The search has been implemented using template driven + activatedRoute method.

File: client/google-search-app/src/app/common/search-form/search-form.component.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  model: any = {};
+
+  constructor(private router: Router) {
+  }
+
+  ngOnInit(): void {
+  }
+
+  search() {
+    console.log(this.model);
```
Comment (karthikeyanpa12) [inline client/google-search-app/src/app/common/search-form/search-form.component.ts:20]: remove console

**Review by karthikeyanpa12 — APPROVED** [review]

**Review by ramsunvtech — APPROVED** [review]

File: client/google-search-app/src/app/app-routing.module.ts [hunk 2]
```diff
@@ -1,10 +1,17 @@
...
+import {NgModule} from '@angular/core';
+import {Routes, RouterModule} from '@angular/router';
+import {HomePageComponent} from "./home-page/home-page.component";
+import {SearchPageComponent} from "./search-page/search-page.component";
 
-const routes: Routes = [];
+const routes: Routes = [
+  {path: '', redirectTo: '/home', pathMatch: 'full'},
+  {path: 'home', component: HomePageComponent},
+  {path: 'search', component: SearchPageComponent},
```
Comment (ramsunvtech) [inline client/google-search-app/src/app/app-routing.module.ts:9]: these routes can be in constants file to group them

File: client/google-search-app/src/app/app.module.ts [hunk 3]
```diff
@@ -1,24 +1,36 @@
...
+    LogoComponent,
+    SearchButtonComponent,
+    SearchFormComponent,
+    LogoComponent
   ],
   imports: [
     BrowserModule,
-    AppRoutingModule
+    AppRoutingModule,
+    FormsModule
```
Comment (ramsunvtech) [inline client/google-search-app/src/app/app.module.ts:30]: we should have module like `HomePageModule` and `SearchPageModule` is the plan in upcoming PRs

File: client/google-search-app/src/app/search-page/search-page.component.ts [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+})
+export class SearchPageComponent implements OnInit {
+
+  constructor(private router: ActivatedRoute) {
+
+  }
+
+  ngOnInit(): void {
+    this.router.queryParamMap.subscribe(d => {
+      console.log(d.get('query'));
```
Comment (ramsunvtech) [inline client/google-search-app/src/app/search-page/search-page.component.ts:17]: remove the console.log here as well

**Review by shangan23 — APPROVED** [review]
Approved

**Review by naveenvignesh5 — APPROVED** [review]

**Pull request merged**